modules/passkey/contracts/SafeWebAuthnSignerFactory.sol

```diff
@@ -42,9 +54,25 @@ contract SafeWebAuthnSignerFactory is ISafeSignerFactory {
         uint256 x,
         uint256 y,
         P256.Verifiers verifiers
-    ) external view override returns (bytes4 magicValue) {
-        if (WebAuthn.verifySignature(message, signature, WebAuthn.USER_VERIFICATION, x, y, verifiers)) {
-            magicValue = ERC1271.MAGIC_VALUE;
+    ) external view override returns (bytes4 /*magicValue*/) {
+        address singleton = SINGLETON;
+        bytes memory data = abi.encodePacked(
+            abi.encodeWithSignature("isValidSignature(bytes32,bytes)", message, signature),
+            x,
+            y,
+            verifiers
+        );
+
+        // solhint-disable-next-line no-inline-assembly
+        assembly {
+            let dataSize := mload(data)
+            let dataLocation := add(data, 0x20)
+
+            let success := staticcall(gas(), singleton, dataLocation, dataSize, 0, 0)
+            if eq(success, 1) {
```

Comment:
Shouldn't this revert when the inner call reverts?

Reply:
I did not add revert because as per specs in `ISafeSignerFactory`. So, thought not reverting would also work.
```
  * @return magicValue Returns the ERC-1271 magic value when the signature is valid. Reverting or
     * returning any other value implies an invalid signature.
```

Reply:
Good point. In this case then, instead of `return`-ing we can copy the return data to offset 0 and with length 32 and then `magicValue := mload(0)`.

This way we don't have separate returning paths from the function, but a single one. Also, from a conceptual level it is nicer in that if the singleton does follow the same return ABI as the factory function, things would still work.

Reply:
done. 
https://github.com/safe-global/safe-modules/pull/370/commits/3138afc05b321142cf4fe22263edd0f94b7bac6e
https://github.com/safe-global/safe-modules/pull/370/commits/7c6bb5755b3be37c5093e85abcd6676fc284d767
https://github.com/safe-global/safe-modules/pull/370/commits/a35771a9833ef023303fe0599587b96246e90b3e